aws/lambda/gateway/variables.tf

```diff
@@ -57,15 +57,32 @@ variable "authorizer" {
 variable "domain" {
   description = "The domain configuration to use for the api gateway."
   type = object({
-    certificate_arn = string
-    zone_name       = string
-    domain          = string
+    # use this without mtls
+    certificate_arn = optional(string)
+    # use this with mtls
+    s3_truststore_uri = optional(string)
+    zone_name          = string
+    domain             = string
   })
   default = null
+
+  validation {
+    # if domain is set, than we need to run an xor validation for either certificate_arn or s3_truststore_uri to be set
+    condition     = (var.domain != null ?
+    ((var.domain.certificate_arn != null && var.domain.s3_truststore_uri== null) ||
+    (var.domain.certificate_arn == null && var.domain.s3_truststore_uri!= null)) : true
+    )
+    error_message = "Either 'certificate_arn' or 's3_truststore_uri' must be set. They cannot both be set or both be null"
+  }
 }
 
 variable "redeployment_trigger" {
   description = "A unique string to force a redeploy of the api gateway. If not set manually, the module will use the configurations for endpoints, api_key, and authorizer config to decide if a redeployment is necessary."
   type        = string
   default     = null
 }
+
+locals {
```

Comment:
I think we always place locals at the top of the file for readability and concistency I would move this up.

Also suggested by terraform [src](https://developer.hashicorp.com/terraform/tutorials/configuration-language/locals) 

> As in any Terraform configuration, the order of your resource definitions and values does not affect how Terraform interprets them. To make your configuration more readable, consider putting local definitions near the top of your files.

Reply:
will do!